resotoworker/resotoworker/config.py

```diff
@@ -1,19 +1,47 @@
+from concurrent.futures import ThreadPoolExecutor, TimeoutError
+from collections import namedtuple
 from resotolib.graph import GraphMergeKind
 from resotolib.config import Config
 from resotolib.proc import num_default_threads
+from resotolib.baseplugin import BaseCollectorPlugin
+from resotolib.logger import log
 from attrs import define, field
-from typing import ClassVar, Optional, List
+from typing import ClassVar, Optional, List, Type
 
 
-def add_config(config: Config) -> None:
+_collector_plugins: List[Type[BaseCollectorPlugin]] = []
+PluginAutoEnabledResult = namedtuple("PluginAutoEnabledResult", ["cloud", "auto_enabled"])
+
+
+def add_config(config: Config, collector_plugins: List[Type[BaseCollectorPlugin]]) -> None:
+    global _collector_plugins
+    _collector_plugins = collector_plugins
```

Comment:
Maybe it is worth moving this to a separate method and calling it before `add_config`?

Reply:
I think add_config is the right place, because we want to ensure that it happens before the config is added. If I move it somewhere else there's a chance for misuse. If it's in here there's no chance it doesn't happen before the config is added.